### PR TITLE
JSX v4 record-based representation for lowercase

### PR DESCRIPTION
**Pull request by mattdamon108**
This PR changes the internal representation of JSX v4 lowercase e.g. `<div />` from object to record

### CODE REVIEW

**Comment by mattdamon108** [comment]
@cristianoc What about the `key` for the props in lowercase? type `JsxDOM.domProps` has a key field as an optional there.

**Comment by mattdamon108** [comment]
It seems fine in v4 sample project. More clean js output.
```js
// as-is
function V4$Baz(props) {
  var name = props.name;
  var name$1 = name !== undefined ? name : "baz";
  var tmp = {};
  if (props.id2 !== undefined) {
    tmp.id = Caml_option.valFromOption(props.id2);
  }
  return React.createElement("div", tmp, name$1, props.children);
}

// to-be
function V4$Baz(props) {
  var name = props.name;
  var name$1 = name !== undefined ? name : "baz";
  return React.createElement("div", {
              id: props.id2
            }, name$1, props.children);
}
```
I'll run on the largest project.

File: tests/ppx/react/expected/lowercases.res.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+@@jsxConfig({version: 4, mode: "classic"})
+
+let _ = ReactDOM.createDOMElementVariadic("div", [])
+let _ = ReactDOM.createDOMElementVariadic("div", ~props={key: "k"}, [])
```
Comment (cristianoc) [inline tests/ppx/react/expected/lowercases.res.txt:4]: How can the same function`createDOMElementVariadic` accept these types?

Comment (mattdamon108) [reply tests/ppx/react/expected/lowercases.res.txt:4]: Not sure I understand your question. The second labeled argument is optional.
```rescript
@variadic @module("react")
external createDOMElementVariadic: (
  string,
  ~props: domProps=?,
  array<React.element>,
) => React.element = "createElement"
```

Comment (mattdamon108) [reply tests/ppx/react/expected/lowercases.res.txt:4]: It reminds me of this https://forum.rescript-lang.org/t/ann-more-strict-checks-in-missed-labels/2117
How is this change going?

**Comment by mattdamon108** [comment]
> I'll run on the largest project.

It works fine on my company project!

**Comment by mattdamon108** [comment]
I think I don't have anything left for this PR now.

**Comment by cristianoc** [comment]
Ready to merge?

**Comment by cristianoc** [comment]
How about updating the changelog?

**Comment by mattdamon108** [comment]
> How about updating the changelog?

Sure, I’ll.

**Comment by mattdamon108** [comment]
> 4d4bb2c

I've added to Polish https://github.com/rescript-lang/syntax/pull/665/commits/4d4bb2cc793699d32a41e829792ef68900c0b05e

**Comment by cristianoc** [comment]
Not sure why suddenly CI gives a compile error on build

**Comment by mattdamon108** [comment]
Let me rebase to the master

**Comment by mattdamon108** [comment]
Build fixed https://github.com/rescript-lang/syntax/pull/665/commits/07190b4957a989f3fbbe10412682da8a7a02884c

**Pull request merged**